### PR TITLE
Added examples and config to docs

### PR DESCRIPTION
**Pull request by oyeb**
## Docs related
* Added a small configuration guide. There's also a broken link to a "Getting Started" guide, which we should fix soon.
* Examples share some context, and can be followed in an `iex` session.

## Tests related
* `commit` and `respond` are now private.
  - Marked a test on `respond` as `:skip` because `respond` is now private, and cannot be tested :(
* Integration tests were failing ever since the lib rename. Fixed that. They were not caught because #13 is still open.

### CODE REVIEW

File: lib/gringotts/gateways/monei.ex [hunk 1]
```diff
@@ -258,13 +359,11 @@ defmodule Gringotts.Gateways.Monei do
      "paymentBrand": card.brand]
   end
 
-  @doc """
-  Makes the request to MONEI's network.
-  """
+  # Makes the request to MONEI's network.
   @spec commit(atom, String.t, keyword, keyword) ::
```
Comment (zeus999) [inline lib/gringotts/gateways/monei.ex]: I think there is no need of  `@spec`, since it is private method.

File: lib/gringotts/gateways/monei.ex [hunk 2]
```diff
@@ -94,10 +148,21 @@ defmodule Gringotts.Gateways.Monei do
...
+  ## Note  
   
   A stand-alone pre-authorization [expires in
   72hrs](https://docs.monei.net/tutorials/manage-payments/backoffice).
+
+  ## Example
+
+  The following session shows how one would (pre) authorize a payment of $40 on the sample `card`.
+
+      iex> auth_result = Gringotts.authorize(:payment_worker, Gringotts.Gateways.Monei, 40, card, opts)
```
Comment (zeus999) [inline lib/gringotts/gateways/monei.ex]: in the example, card parameter is not defined and same for opts.
The example should mimic the same behaviour as someone is going to use it.

card = %CreditCard{}
opts = []
amount = 40

iex> auth_result = Gringotts.authorize(:payment_worker, Gringotts.Gateways.Monei, amount, card, opts)

Comment (oyeb) [reply lib/gringotts/gateways/monei.ex]: Please refer lines [163](https://github.com/aviabird/gringotts/pull/38/files#diff-cd577a67abd155af0d2d475f678770c1R163) (right below) and [85](https://github.com/aviabird/gringotts/pull/38/files#diff-cd577a67abd155af0d2d475f678770c1R85).

File: lib/gringotts/gateways/monei.ex [hunk 3]
```diff
@@ -33,14 +38,63 @@ defmodule Gringotts.Gateways.Monei do
...
-  ## Caveats
+  | Config parameter | MONEI secret   |
+  | -------          | ----           |
+  | `:userId`        | **User ID**    |
+  | `:entityId`      | **Channel ID** |
+  | `:password`      | **Password**   |
+
+  Your Application config **must include the `:userId`, `:entityId`, `:password`
+  fields** and would look something like this:
+
```
Comment (SagarKarwande) [inline lib/gringotts/gateways/monei.ex]: Replace Kuber.Hex namespace with Gringotts

File: lib/gringotts/gateways/monei.ex (hunk 1 above)
Comment (oyeb) [reply lib/gringotts/gateways/monei.ex]: No harm in documenting though, it does not end up in the docs, and is for devs who happen to read this file.

File: lib/gringotts/gateways/monei.ex (hunk 3 above)
Comment (oyeb) [reply lib/gringotts/gateways/monei.ex]: Done.

File: lib/gringotts/gateways/monei.ex (hunk 2 above)
Comment (zeus999) [reply lib/gringotts/gateways/monei.ex]: ideally this should be present before each example, as it helps developers to understand the examples better.

Comment (oyeb) [reply lib/gringotts/gateways/monei.ex]: Done!

**Pull request merged**